platforms/software/dependency-management/src/main/java/org/gradle/api/internal/artifacts/configurations/TasksFromProjectDependencies.java

```diff
@@ -44,11 +51,13 @@ public void visitDependencies(TaskDependencyResolveContext context) {
         taskDependencyDelegate.visitDependencies(context);
     }
 
-    void resolveProjectDependencies(TaskDependencyResolveContext context, Set<ProjectDependency> projectDependencies) {
+    void resolveProjectDependencies(TaskDependencyResolveContext context, Set<ProjectDependency> projectDependencies, ProjectStateRegistry projectStateRegistry) {
```

Comment:
❗ Non `private static` methods called from constructor of a non-`final` class are trouble.

The only field used here is `taskName`.  Which is only read in a test.  Can this field be deleted completely and just passed here as an argument?

Reply:
Made this private & static. Removed the field. Deleted the test since it really didn't test anything